core/integration/module_cli_test.go

```diff
@@ -1079,3 +1079,251 @@ func (m *OtherObj) FnE() *dagger.Container {
+				With(daggerExec("init", "--sdk=go", "--name=bar", "--source=.")).
+				WithWorkdir("/work").
+				With(daggerExec("init", "--sdk=go", "--name=foo", "--source=.")).
+				With(daggerExec("install", "./bar"))
+
+			daggerjson, err := ctr.File("dagger.json").Contents(ctx)
+			require.NoError(t, err)
+			require.Contains(t, daggerjson, "bar")
+
+			daggerjson, err = ctr.
+				WithoutDirectory("/work/bar").
+				With(daggerExec("uninstall", "bar")).
+				File("dagger.json").Contents(ctx)
+			require.NoError(t, err)
+			require.NotContains(t, daggerjson, "bar")
+		})
+
+		t.Run("dependency source is removed before calling uninstall using relative path", func(ctx context.Context, t *testctx.T) {
+			c := connect(ctx, t)
+
+			ctr := c.Container().
+				From(alpineImage).
+				WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+				WithWorkdir("/work/bar").
+				With(daggerExec("init", "--sdk=go", "--name=bar", "--source=.")).
+				WithWorkdir("/work").
+				With(daggerExec("init", "--sdk=go", "--name=foo", "--source=.")).
+				With(daggerExec("install", "./bar"))
+
+			daggerjson, err := ctr.File("dagger.json").Contents(ctx)
+			require.NoError(t, err)
+			require.Contains(t, daggerjson, "bar")
+
+			daggerjson, err = ctr.
+				WithoutDirectory("/work/bar").
+				With(daggerExec("uninstall", "./bar")).
+				File("dagger.json").Contents(ctx)
+			require.NoError(t, err)
+			require.NotContains(t, daggerjson, "bar")
+		})
+	})
+
+	t.Run("git dependency", func(ctx context.Context, t *testctx.T) {
+		t.Run("uninstall a dependency configured in dagger.json with version number", func(ctx context.Context, t *testctx.T) {
+			c := connect(ctx, t)
+
+			ctr := c.Container().
+				From(alpineImage).
+				WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+				WithWorkdir("/work").
+				With(daggerExec("init", "--sdk=go", "--name=foo", "--source=.")).
+				With(daggerExec("install", "github.com/shykes/daggerverse/hello@v0.3.0"))
+
+			daggerjson, err := ctr.File("dagger.json").Contents(ctx)
+			require.NoError(t, err)
+			require.Contains(t, daggerjson, "hello")
+
+			daggerjson, err = ctr.With(daggerExec("uninstall", "github.com/shykes/daggerverse/hello@v0.3.0")).
+				File("dagger.json").Contents(ctx)
+			require.NoError(t, err)
+			require.NotContains(t, daggerjson, "hello")
+		})
+
+		t.Run("uninstall a dependency configured in dagger.json without version number", func(ctx context.Context, t *testctx.T) {
+			c := connect(ctx, t)
+
+			ctr := c.Container().
+				From(alpineImage).
+				WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+				WithWorkdir("/work").
+				With(daggerExec("init", "--sdk=go", "--name=foo", "--source=.")).
+				With(daggerExec("install", "github.com/shykes/daggerverse/hello"))
+
+			daggerjson, err := ctr.File("dagger.json").Contents(ctx)
+			require.NoError(t, err)
+			require.Contains(t, daggerjson, "hello")
+
+			daggerjson, err = ctr.With(daggerExec("uninstall", "github.com/shykes/daggerverse/hello@v0.3.0")).
```

Comment:
Is this version ignored, or does the version have to match? If the latter, won't this test fail if a new version of `hello` ships?

Reply:
Yeah agreed on both counts - definitely an edge case so I'm not too bothered by the current behavior, but it doesn't seem like it should be ignored if it's specified

Reply:
I've adjusted the code and tests to check for specific version if version number is specified. 

As part of this I also refactored the testcases (for `uninstalling git dependency` relates tests) to use test-matrix instead of individual tests that makes them significantly more readable. 

I'll try to get this done for `local dependency` testcases as well. 

in the meantime it would be great to receive any additional feedback that you might have.